src/lib.rs

```diff
@@ -699,7 +712,12 @@ pub struct Lines {
 }
 
 impl Lines {
+    pub const MAX_LINES: usize = ffi::GPIOHANDLES_MAX;
+
     fn new(chip: Arc<Box<InnerChip>>, offsets: &[u32]) -> Result<Lines> {
+        if offsets.len() == 0 || offsets.len() > Lines::MAX_LINES {
```

Comment:
nit: `offsets.len() > 0` - Arguably slightly more idiomatic to use `offsets.is_empty()`.  Not really worth changing.

Reply:
Yeah, I cringed a little at the look of calling it twice in the same line, but I thought it more clearly showed intent (whether it was out of range). I thought to cache it in a var:
```
let n = offsets.len();
if n == 0 || n > Lines::MAX_LINES {
```
But, overall, I'm fairly agnostic.